sw/device/tests/sram_ctrl_sleep_sram_ret_contents_test.c

```diff
@@ -100,22 +190,26 @@ bool test_main(void) {
           kDifToggleEnabled));
 
       CHECK_DIF_OK(dif_aon_timer_wakeup_stop(&aon_timer));
-      aon_timer_testutils_watchdog_config(&aon_timer, 0xffffffff, 10, false);
 
       // Enter low power mode.
       LOG_INFO("Issue WFI to enter sleep");
+      aon_timer_testutils_watchdog_config(&aon_timer, UINT32_MAX, 20, false);
       wait_for_interrupt();
     } else {
       LOG_INFO("Watchdog reset");
-      // Check that the previously written retention SRAM data can still
-      // be read successfully.
-      retention_sram_check(false);
+      // Check that the previously written retention SRAM data cannot be read
+      // successfully.
+      retention_sram_check(
+          (check_config_t){.do_write = false, .is_equal = false});
     }
   } else {
     LOG_FATAL("Unexepected reset type detected.");
   }
 
   // Turn off the AON timer hardware completely before exiting.
   aon_timer_testutils_shutdown(&aon_timer);
+  if (kDeviceType == kDeviceSimDV) {
```

Comment:
Is there a particular reason this is needed in sim DV?

Reply:
@weicaiyang is there anything else that we need to disable so that the test can exit cleanly in DV?

Reply:
@weicaiyang @cindychip @sriyerg is there anything else that we need to disable so that the test can exit cleanly in DV?

Reply:
I opened an issue to investigate why this is needed so that we can make progress and merge this: #16597

Reply:
AFAICT, the aon timer was the only one. Unserviced alerts also caused EOT failures, but I don't think this test is expected to generate an alert, so I am not sure what else could be causing a hold up. 

This was added in the other test in this PR: https://github.com/lowRISC/opentitan/pull/16232. There does not appear to be any comment explaining why it was added. @engdoreis, do you recall what the error syndrome was without the reset?

Reply:
This was added in the PR https://github.com/lowRISC/opentitan/pull/16232, because it was in the previous retention sram test, and the PR https://github.com/lowRISC/opentitan/pull/16232 merged it with the main sram test.
Without the reset the testbench would block until timeout.